app/frontend/src/features/donations/DonationsBoxMixed.tsx

```diff
@@ -0,0 +1,440 @@
+    marginBottom: theme.spacing(2),
+  },
+
+  inputWrapper: {
+    position: "relative",
+    "&::before": {
+      content: `'${DONATIONSBOX_CURRENCY}'`,
+      position: "absolute",
+      top: "50%",
+      transform: "translateY(-50%)",
+      left: theme.spacing(1),
+      color: theme.palette.grey[600],
+      fontWeight: 700,
+      fontSize: theme.typography.button.fontSize,
+    },
+  },
+
+  inputNumber: {
+    width: "100%",
+    height: "100%",
+    border: `2px solid ${theme.palette.grey[200]}`,
+    paddingLeft: theme.spacing(2),
+    color: theme.palette.grey[600],
+    fontWeight: 700,
+    fontSize: theme.typography.button.fontSize,
+    "&&": {
+      borderRadius: theme.shape.borderRadius * 2,
+      boxShadow: "initial",
+    },
+    "&:hover": {
+      border: `2px solid ${theme.palette.primary.main}`,
+    },
+    "&:focus-visible": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      outline: "none",
+      boxShadow: "none",
+    },
+  },
+
+  inputNumberActive: {
+    "&&": {
+      border: `2px solid ${theme.palette.primary.main}`,
+    },
+  },
+}));
+
+export interface DonationFormData {
+  amount: number;
+  recurring: boolean;
+}
+
+export default function DonationsBoxMixed() {
+  const classes = useStyles();
+
+  const [success, setSuccess] = useState(false);
+  const [cancelled, setCancelled] = useState(false);
+
+  const location = useLocation();
+
+  useEffect(() => {
+    const query = new URLSearchParams(location.search);
+    setSuccess(query.get("success") ? true : false);
+    setCancelled(query.get("cancelled") ? true : false);
+  }, [location]);
+
+  const {
+    control,
+    register,
+    handleSubmit,
+    reset: resetForm,
+    errors,
+    watch
+  } = useForm<DonationFormData>({ defaultValues: { recurring: true } });
+
+  const watchedAmount = watch("amount");
+
+  const {
+    error,
+    isLoading,
+    mutate: initiateDonation,
+  } = useMutation<void, GrpcError, DonationFormData>(
+    async (formData) => {
+      const stripe = (await stripePromise)!;
+      const session_id = await service.donations.initiateDonation(formData);
+      // When the customer clicks on the button, redirect them to Checkout.
+      const result = await stripe.redirectToCheckout({
+        sessionId: session_id,
+      });
+      if (result.error) {
+        throw Error(result.error.message);
+      }
+    },
+    {
+      onSuccess: () => {
+        resetForm();
+      },
+    }
+  );
+
+  const onSubmit = handleSubmit((data) => {
+    initiateDonation(data);
+  });
+
+  return (
+    <>
+      <form onSubmit={onSubmit} className={classes.donationsBox}>
+        {error && <Alert severity="error">{error.message}</Alert>}
+        {success && (
+          <Alert severity="success">
+            Thank you for donating to Couchers.org! We appreciate your support!
+          </Alert>
+        )}
+        {cancelled && (
+          <Alert severity="warning">The payment was cancelled.</Alert>
```

Comment:
Put messages here in constants

Reply:
Not sure about how to handle the type issue, just give a look to that in the final commit :)